distributed/scheduler.py

```diff
@@ -3466,9 +3473,11 @@ def _add_to_memory(
             s.discard(ts)
             if not s and not dts.who_wants:
                 recommendations[dts.key] = "released"
+                self.task_net_nbytes[ts.prefix] -= dts.nbytes
```

Comment:
Should we divide by the number of dependents a dependency has?

If a large dependency has 100 dependents, they probably shouldn't all think that they're going to release a bunch of memory.

Reply:
Let me describe what I hoped would happen, which may not be how I coded / understood things. If you have a graph like

```python
  
        a1    a2
       /  \  /  \
     b0    b1    b2
       \  /  \  /
        c1    c2
  
  ```

then `a1` can be freed when both b0 and b1 are done. Let's suppose b0 finishes before b1. Then, I think the way I've written it things go like

1. b0 finishes. The scheduler sees that the worker running `b1` still wants `a0`, so we don't even reach this section, there aren't any dependents to free. `task_net_bytes['b']` is still 0
2. b1 finishes. The scheduler sees that `a1` can now be freed, so `tast_net_bytes['b']` is decremented by `sizeof(a1)`.